libs/blocks/merch/merch.js

```diff
@@ -40,70 +27,88 @@ function omitNullValues(target) {
   return target;
 }
 
-const getTacocatEnv = (envName, locale) => {
-  const wcsLocale = GEO_MAPPINGS[locale.prefix] ?? locale.ietf;
-  // eslint-disable-next-line prefer-const
-  let [language, country = 'US'] = wcsLocale.split('-', 2);
+const imsCountryPromise = new Promise((resolve) => {
+  let count = 0;
+  const check = setInterval(() => {
+    count += 1;
+    if (window.adobeIMS) {
+      clearInterval(check);
+      if (window.adobeIMS.isSignedInUser()) {
+        window.adobeIMS
+          .getProfile()
+          .then(({ countryCode }) => resolve(countryCode))
+          .catch(() => resolve());
+      } else {
+        resolve();
+      }
+    } else if (count > 25) {
+      clearInterval(check);
+      resolve();
+    }
+  }, 200);
+});
+
+export const getTacocatEnv = (envName, locale) => {
+  const wcsLocale = (GEO_MAPPINGS[locale.prefix] ?? locale.ietf).split('-', 2);
+  let language = wcsLocale[0];
+  const country = wcsLocale[1] || 'US';
   if (!SUPPORTED_LANGS.includes(language)) {
     language = 'en';
   }
-  const host = envName === ENV_PROD
-    ? 'https://www.adobe.com'
-    : 'https://www.stage.adobe.com';
-  const scriptUrl = `${host}/special/tacocat/lib/${VERSION}/tacocat.js`;
+  const host = envName === ENV_LOCAL
```

Comment:
This means that every environment that is not `local` will default to `prod`, hence `stage` would be considered `prod` as well, no?

Reply:
@honstar yes, the reason behind this is the following:

When we test the "after URL" in the PRs, the environment is considered stage. If we load tacocat from stage environment, as soon as PR is merged, it will load from prod environment, which means we would have never tested with the prod url.

I consider tacocat stage more for tacocat devs to validate things while developing locally.

I'm still not 100% sure whether this is a good decision though. WDYT?

Reply:
If I'm not mistaken, you can force override the environment to be used by adding a query parameter `env`, see https://github.com/adobecom/milo/blob/main/libs/utils/utils.js#L110.

Reply:
Indeed this comes handy. Let me update this part, thanks.

Reply:
@honstar  reverted back to prod/stage.